apps/backtest-report/components/TradingViewChart.js

```diff
@@ -398,14 +398,18 @@ const TradingViewChart = (props) => {
       if (chartData.positionHistory) {
         const lineSeries = chart.current.addLineSeries();
         const costLine = positionAverageCostHistoryToLineData(currentInterval, chartData.positionHistory);
-        lineSeries.setData(costLine);
+        const costLineIdx = {}
+        const costLineDeduplicated = costLine.filter( v => { const exists = costLineIdx[v.time]; costLineIdx[v.time] = v.value; return !exists  });
```

Comment:
move this check to the positionAverageCostHistoryToLineData function? there is already a for loop.

Reply:
sure. let me look into the function